cmd/snapd-apparmor/main.go

```diff
@@ -149,7 +149,7 @@ func loadAppArmorProfiles() error {
 
 func isContainer() bool {
 	// systemd's implementation may fail on WSL2 with custom kernels
-	return release.OnWSL || (exec.Command("systemd-detect-virt", "--quiet", "--container").Run() == nil)
+	return release.OnWSL || systemd.IsContainer()
```

Comment:
there are 2 more call locations: osutil/squashfs/fstype.go and daemon/api_general.go.

Reply:
Yes I know.

- squashfs cannot import systemd, I've mentioned that in the comment above.
- daemon doesn't check for containers so I left it out, the signature would be `systemd.VirtualizationType` but this is orthogonal to this fix.